d3rlpy/torch_utility.py

```diff
@@ -388,6 +391,19 @@ def reset_optimizer_states(self) -> None:
             if isinstance(v, torch.optim.Optimizer):
                 v.state = collections.defaultdict(dict)
 
+    def get_torch_modules(self) -> List[nn.Module]:
+        torch_modules: List[nn.Module] = []
+        for v in asdict_without_copy(self).values():
+            if isinstance(v, nn.Module):
+                torch_modules.append(v)
+        return torch_modules
+
+    def get_gradients(self) -> Iterator[Tuple[str, Float32NDArray]]:
+        for module in self.get_torch_modules():
```

Comment:
Concatenate module name and parameter names otherwise names conflict.
```py
for module_name, module in self.get_torch_modules().items():
...
        yield f"{module_name}.{name}", parameter.grad.cpu().detach().numpy()
```

Reply:
5b8bb3c